x-pack/plugins/infra/public/containers/logs/log_source/log_source.ts

```diff
@@ -133,53 +109,67 @@ export const useLogSource = ({
     [resolvedSourceConfiguration]
   );
 
-  const isLoadingSourceConfiguration = useMemo(
-    () => loadSourceConfigurationRequest.state === 'pending',
-    [loadSourceConfigurationRequest.state]
-  );
-
-  const isUpdatingSourceConfiguration = useMemo(
-    () => updateSourceConfigurationRequest.state === 'pending',
-    [updateSourceConfigurationRequest.state]
-  );
-
-  const isLoadingSourceStatus = useMemo(() => loadSourceStatusRequest.state === 'pending', [
-    loadSourceStatusRequest.state,
-  ]);
-
-  const isLoading = useMemo(
-    () => isLoadingSourceConfiguration || isLoadingSourceStatus || isUpdatingSourceConfiguration,
-    [isLoadingSourceConfiguration, isLoadingSourceStatus, isUpdatingSourceConfiguration]
-  );
-
-  const isUninitialized = useMemo(
-    () =>
-      loadSourceConfigurationRequest.state === 'uninitialized' ||
-      loadSourceStatusRequest.state === 'uninitialized',
-    [loadSourceConfigurationRequest.state, loadSourceStatusRequest.state]
-  );
-
-  const hasFailedLoadingSource = useMemo(
-    () => loadSourceConfigurationRequest.state === 'rejected',
-    [loadSourceConfigurationRequest.state]
-  );
-
-  const hasFailedLoadingSourceStatus = useMemo(() => loadSourceStatusRequest.state === 'rejected', [
-    loadSourceStatusRequest.state,
-  ]);
-
-  const loadSourceFailureMessage = useMemo(
-    () =>
-      loadSourceConfigurationRequest.state === 'rejected'
-        ? `${loadSourceConfigurationRequest.value}`
-        : undefined,
-    [loadSourceConfigurationRequest]
+  const isLoadingSourceConfiguration = loadSourceConfigurationRequest.state === 'pending';
+  const isResolvingSourceConfiguration = resolveSourceConfigurationRequest.state === 'pending';
+  const isLoadingSourceStatus = loadSourceStatusRequest.state === 'pending';
+  const isUpdatingSourceConfiguration = updateSourceConfigurationRequest.state === 'pending';
+
+  const isLoading =
+    isLoadingSourceConfiguration ||
+    isResolvingSourceConfiguration ||
+    isLoadingSourceStatus ||
+    isUpdatingSourceConfiguration;
+
+  const isUninitialized =
+    loadSourceConfigurationRequest.state === 'uninitialized' ||
+    resolveSourceConfigurationRequest.state === 'uninitialized' ||
+    loadSourceStatusRequest.state === 'uninitialized';
+
+  const hasFailedLoadingSource = loadSourceConfigurationRequest.state === 'rejected';
+  const hasFailedResolvingSource = resolveSourceConfigurationRequest.state === 'rejected';
+  const hasFailedLoadingSourceStatus = loadSourceStatusRequest.state === 'rejected';
+
+  const latestLoadSourceFailures = [
+    loadSourceConfigurationRequest,
+    resolveSourceConfigurationRequest,
+    loadSourceStatusRequest,
+  ]
+    .filter(isRejectedPromiseState)
+    .map(({ value }) => (value instanceof Error ? value : new Error(`${value}`)));
+
+  const hasFailedLoading = latestLoadSourceFailures.length > 0;
+
+  const loadSource = useCallback(async () => {
```

Comment:
This should now be the primary trigger to load the source config, because it properly orchestrates the calls.